### ISSUE TITLE
Whiskey hangs when typing cmd + up

### ISSUE THREAD

**Issue by kylef** [nothingmagical/whiskey-issues#68]
### Steps to reproduce

1. Open a new Whiskey document
2. Type `---`
3. Hit `cmd` + (`up`)

### Expected Result

Cursor to go to index 0.

### Actual result

Whiskey seems to hang. I forced quit a few times and it still hangs.

---

Eventually it crashed (could be related or unrelated):

```
Process:               Whiskey [18466]
Path:                  /Applications/Whiskey.app/Contents/MacOS/Whiskey
Identifier:            com.nothingmagical.whiskey
Version:               0.3.7 (49)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Whiskey [18466]
User ID:               501

Date/Time:             2015-01-16 17:01:12.795 +0100
OS Version:            Mac OS X 10.10.1 (14B25)
Report Version:        11
Anonymous UUID:        D258F0F4-72DF-AB4C-0C3A-8424756F417C

Sleep/Wake UUID:       25ACE2AD-010C-4CFA-9B4F-E832D9130C9F

Time Awake Since Boot: 53000 seconds
Time Since Wake:       12 seconds

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Application Specific Information:
*** Terminating app due to uncaught exception 'NSRangeException', reason: '-[NSTaggedPointerString substringWithRange:]: Range {18446744073709551615, 1} out of bounds; string length 4'
terminating with uncaught exception of type NSException
abort() called

Application Specific Backtrace 1:
0   CoreFoundation                      0x00007fff956ee64c __exceptionPreprocess + 172
1   libobjc.A.dylib                     0x00007fff8aece6de objc_exception_throw + 43
2   CoreFoundation                      0x00007fff956ee4fd +[NSException raise:format:] + 205
3   CoreFoundation                      0x00007fff9569658f -[NSTaggedPointerString substringWithRange:] + 239
4   MarkdownKit                         0x000000010725d9b0 -[MDKParser parseUnderlinedHeaderWithScanner:parent:range:] + 332
5   MarkdownKit                         0x000000010725c1b5 -[MDKParser parseBlockLevel:offset:] + 2923
6   MarkdownKit                         0x000000010725b541 -[MDKParser parseString:] + 237
7   WhiskeyCore                         0x000000010728faac -[WHYTextStorage highlightRange:] + 89
8   WhiskeyCore                         0x000000010728fa14 -[WHYTextStorage setNeedsHighlight] + 473
9   CoreFoundation                      0x00007fff956aacbc __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
10  CoreFoundation                      0x00007fff9559c1b4 _CFXNotificationPost + 3140
11  CoreFoundation                      0x00007fff95605db5 CFNotificationCenterPostNotification + 117
12  CoreFoundation                      0x00007fff956caa4f _CFPrefsValuesChangedExternally + 79
13  libdispatch.dylib                   0x00007fff95aef323 _dispatch_call_block_and_release + 12
14  libdispatch.dylib                   0x00007fff95aeac13 _dispatch_client_callout + 8
15  libdispatch.dylib                   0x00007fff95af6cbf _dispatch_main_queue_callback_4CF + 861
16  CoreFoundation                      0x00007fff95641c59 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
17  CoreFoundation                      0x00007fff955fe2ef __CFRunLoopRun + 2159
18  CoreFoundation                      0x00007fff955fd838 CFRunLoopRunSpecific + 296
19  HIToolbox                           0x00007fff8b8af43f RunCurrentEventLoopInMode + 235
20  HIToolbox                           0x00007fff8b8af1ba ReceiveNextEventCommon + 431
21  HIToolbox                           0x00007fff8b8aeffb _BlockUntilNextEventMatchingListInModeWithFilter + 71
22  AppKit                              0x00007fff97ce36d1 _DPSNextEvent + 964
23  AppKit                              0x00007fff97ce2e80 -[NSApplication nextEventMatchingMask:untilDate:inMode:dequeue:] + 194
24  AppKit                              0x00007fff97cd6e23 -[NSApplication run] + 594
25  AppKit                              0x00007fff97cc22d4 NSApplicationMain + 1832
26  Whiskey                             0x000000010719fce1 Whiskey + 171233
27  libdyld.dylib                       0x00007fff953ac5c9 start + 1
28  ???                                 0x0000000000000001 0x0 + 1
```

**Comment by soffes** [comment]
This should be fixed in the next version of the parser. I'll add a test for it though. Thanks!

**Comment by soffes** [comment]
Fixed in nothingmagical/whiskey@be648a80e4d5537605f6017109049292a2a16fcc
